OTAnalytics/application/state.py

```diff
@@ -404,6 +404,22 @@ def _update_image(self) -> None:
         self._track_view_state.background_image.set(self._plotter.plot())
 
 
+class VideosMetadata:
+    def __init__(self) -> None:
+        self._metadata: list[VideoMetadata] = []
+
+    def update(self, metadata: VideoMetadata) -> None:
+        self._metadata.append(metadata)
+
+    @property
+    def first_video_start(self) -> Optional[datetime]:
```

Comment:
This assumes that `self._metadata` is sorted by `VideoMetadata.recorded_start_date`. This might not always be the case, especially when  loading tracks in no particular order, right?